compiler/src/dotty/tools/dotc/transform/ElimRepeated.scala

```diff
@@ -91,12 +96,10 @@ class ElimRepeated extends MiniPhase with InfoTransformer { thisPhase =>
   private def seqToArray(tree: Tree, pt: Type)(implicit ctx: Context): Tree = tree match {
     case SeqLiteral(elems, elemtpt) =>
       JavaSeqLiteral(elems, elemtpt)
-    case app@Apply(fun, args) if defn.WrapArrayMethods().contains(fun.symbol) => // rewrite a call to `wrapXArray(arr)` to `arr`
```

Comment:
Why was this optimization dropped?

Reply:
It is not needed anymore. When calling into java repeated we used to insert an implicit conversion from `Array` to `Seq` that we later reverted. For example:
```scala
def test(xs: Array[String]) = Paths.get("Hello", xs: _*)
```
Before this path:
```scala
// after frontend:
def test(xs: Array[String]) = Paths.get("Hello", wrapRefArray[String](xs):String*)

// after elimRepeated
def test(xs: Array[String]): = Paths.get("Hello", xs)
```
After this patch:
```scala
// after frontend:
def test(xs: Array[String]) = Paths.get("Hello", xs:String*) // no more call to wrapRefArray

// after elimRepeated
def test(xs: Array[String]): = Paths.get("Hello", xs)
```

Reply:
Thanks for explaining!